### ISSUE TITLE
Support translate_moe model from fairseq.

### ISSUE THREAD

**Issue by BrightXiaoHan** [OpenNMT/CTranslate2#783]
Hi! I have train a MOE (Mix of expert) model following [this example of moe](https://github.com/pytorch/fairseq/blob/main/examples/translation_moe/README.md).

 It's possible to convert this trained model to ctranslate2 format?

Thanks!

**Comment by guillaumekln** [comment]
I browsed the MOE source code and I think the model is compatible with CTranslate2, although I did not find a pretrained model to verify it.

The conversion should include 2 additional options:

```text
--user_dir /path/to/fairseq/examples/translation_moe/translation_moe_src --no_default_special_tokens 
```

* `--user_dir` is required to register the custom MOE task (note that the option has been added very recently in #782 is not yet released in a version)
* `--no_default_special_tokens` is required because the decoder start token is used to select the expert. In that case you should manually add the EOS token to the source and set the expert token in the target prefix, e.g.:

```python
translator.translate_batch([tokens + ["</s>"]], target_prefix=[["<expert_0>"]])
```

**Comment by BrightXiaoHan** [comment]
Oh, thanks!  I will try it out.

**Comment by guillaumekln** [comment]
The `--user_dir` option is available in the latest version (2.16.0). Can you try the conversion as shown above and update this issue?

**Comment by guillaumekln** [comment]
I'm closing this issue. Feel free to reopen if it is not working as expected.
